File: src/network/packet/impl/response/client/CommandExecuteResponsePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\response\client;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\data\ServerCommandExecutionResult;
use pocketcloud\cloud\network\packet\ResponseClientPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class CommandExecuteResponsePacket extends ResponseClientPacket {

    public function __construct(private ?ServerCommandExecutionResult $commandExecutionResult = null) {}

    public function handle(ServerClient $client): void {
        if (($server = $client->getServer()) !== null) {
            $server->handleCommandResponse($this->commandExecutionResult);
        }
    }

    public function decodePayload(PacketData $packetData): void {
        $packetData->readAllTypeSafe([&$this->commandExecutionResult], [fn() => $packetData->readServerCommandExecutionResult()]);
    }

    public function getCommandExecutionResult(): ?ServerCommandExecutionResult {
        return $this->commandExecutionResult;
    }

    public static function create(ServerCommandExecutionResult $commandExecutionResult): self {
        return new self($commandExecutionResult);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/response/client/CommandExecuteResponsePacket.java
package de.pocketcloud.cloud.network.packet.impl.response.client;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ResponseClientPacket;
import de.pocketcloud.cloud.network.packet.type.ServerCommandExecutionResult;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class CommandExecuteResponsePacket extends ResponseClientPacket {

    private ServerCommandExecutionResult commandExecutionResult;

    public CommandExecuteResponsePacket(ServerCommandExecutionResult commandExecutionResult) {
        this.commandExecutionResult = commandExecutionResult;
    }

//    @Override
//    public void handle(@NotNull ServerClient client) {
//        var server = client.server();
//        if (server != null) {
//            CloudServersHandler.handleCommandResponse(server, commandExecutionResult);
//        }
//    }


    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void decodePayload(PacketData packetData) {
        this.commandExecutionResult = ServerCommandExecutionResult.read(packetData.readMap());
    }

    public static CommandExecuteResponsePacket create(ServerCommandExecutionResult commandExecutionResult) {
        return new CommandExecuteResponsePacket(commandExecutionResult);
    }
}